tests/Repository.py

```diff
@@ -126,6 +126,7 @@ def testEditWithAllArguments(self):
             allow_squash_merge=True,
             allow_merge_commit=True,
             allow_rebase_merge=True,
+            delete_branch_on_merge=True,
```

Comment:
My only comment here would be we don't actually assert on any of the changed parameters. To help increase coverage, can you ``assertTrue delete_branch_on_merge`` and add a FIXME comment to assert the rest?

Reply:
Done. 

Ironically, adding an assertion for this parameter found a recursion bug due to a copy / paste error :)

Reply:
Untested code is broken code. :-)